main/lsp/LSPPreprocessor.h

```diff
@@ -109,6 +108,11 @@ class LSPPreprocessor final {
      */
     void preprocessAndEnqueue(QueueState &state, std::unique_ptr<LSPMessage> msg, absl::Mutex &stateMtx);
 
+    /**
+     * Clears the undo history. Used in single-threaded mode to signal when updates are safely committed.
```

Comment:
is it possible to make single threaded mode less special?

Reply:
Actually, I might not need this function anymore. In single threaded mode, we should clear out the undo history the next time an edit comes in.